common/djangoapps/third_party_auth/models.py

```diff
@@ -204,6 +212,14 @@ def get_authentication_backend(self):
         """Gets associated Django settings.AUTHENTICATION_BACKEND string."""
         return '{}.{}'.format(self.backend_class.__module__, self.backend_class.__name__)
 
+    @property
+    def display_for_login(self):
+        """
+        Determines whether the provider ought to be shown as an option with
+        which to log in on the login screen, registration screen, and elsewhere.
+        """
+        return bool(self.enabled and self.accepts_logins and self.visible)
```

Comment:
Not strictly speaking; this is more of a peace-of-mind thing. Boolean operators in Python return that value which confirms the statement as either True-y or False-y, so when you do `a or b or c`, you're not necessarily getting a boolean value, but either the first of a, b, or c that's truthy, or c if all are falsey.

Of course, we're working with model attributes here, so there's no reason to expect any of the values in the statement to not be bools already. In this particular case, it doesn't really make a difference, so I'd be happy to remove the explicit cast here if you'd like.